lib/TokenManager.js

```diff
@@ -140,18 +140,30 @@ function renew(sdk, tokenMgmtRef, storage, key) {
   // Store the renew promise state, to avoid renewing again
   if (!tokenMgmtRef.renewPromise[key]) {
     tokenMgmtRef.renewPromise[key] = sdk.token.renew(token)
-    .then(function(freshToken) {
-      if (!get(storage, key)) {
-        // It is possible to enter a state where the tokens have been cleared
-        // after a renewal request was triggered. To ensure we do not store a
-        // renewed token, we verify the promise key doesn't exist and return.
-        return;
+    .then(function(freshTokens) {
+      // We may receive more tokens than we requested
+      var map = {};
+      if (freshTokens instanceof Array === false) {
+        freshTokens = [freshTokens];
       }
-      add(sdk, tokenMgmtRef, storage, key, freshToken);
-      tokenMgmtRef.emitter.emit('renewed', key, freshToken, token);
+      freshTokens.forEach(function(freshToken) {
+        var inferredKey = freshToken.idToken ? 'idToken' : freshToken.accessToken ? 'accessToken' : key;
+        map[inferredKey] = freshToken;
+
+        var oldToken = get(storage, inferredKey);
```

Comment:
nit: This is a duplicate call. You can use the `token` variable above instead.

Reply:
If you look at the code being replaced, this extra call was there before, I believe it is to prevent an error where the token existed before refresh attempt but was cleared while the request was in flight.  In the case that the token may have changed while the request was in flight, I am taking the freshest "old" token